### ISSUE TITLE
Edit data doesnt work once you revert an update

### ISSUE THREAD

**Issue by akamath46297** [Microsoft/sqlopsstudio#534]
- SQL Operations Studio Version: 0.25.2

Steps to Reproduce:
Click on edit data to edit a table
Update cells of a particular row and then switch rows so that the update gets commited
Now update cells of some other row and switch cells within the row
Right click on the row and click revert row
The row gets reverted
Try updating the rows again

Expected:
You should be able to update the rows

Actual:
sqlops doesn't track the updates to any rows after revert

**Comment by akamath46297** [comment]
There is another issue:
When the edit data view is open and if tab out to another window and come back to edit data window, updates don't work anymore

**Comment by kevcunnane** [comment]
@benrr101 are these related to fixes you made in Edit Data? These are potentially serious scenario blockers so we should prioritize investigating this.

**Comment by benrr101** [comment]
@kevcunnane This isn't very serious, the simple workaround for the both issues is to click refresh.
They're definitely bugs, tho. At this point it's unclear whether they are directly caused by the fixes I made. I'll investigate.

**Comment by akamath46297** [comment]
@benrr101 , I see that sqlops is not raising events which makes call to the toolsservice after switching tabs

**Comment by benrr101** [comment]
For issue 2, it looks like the onActiveCellChanged event loses its subscription once the tab changes. I'll need to confirm with @anthonydresser, tho.

For issue 1, I'm not entirely certain yet. However, I noticed that the revert row code is unnecessarily calling edit/commit (which has been in since the beginning). This likely isn't causing the issue, but it's something that we should address.

**Comment by benrr101** [comment]
@akamath46297 Update on this: For issue 2, I've made a change to the Microsoft/angular2-slickgrid repo to expose the onActiveCellChanged event. A small change also needs to be made to Sql Ops to take these changes, but it solves the issue and removes a hacky solution we used before.

The same code review will also remove the call to commit when a row is reverted. As for the unable to edit after revert, I'll address that with a separate PR.
